ui/src/components/PolicyEditor.js

```diff
@@ -0,0 +1,226 @@
+  return {
+    timePolicy: _.first(policyParsed),
+    aggFunctions: _.chain(policyParsed)
+      .nth(1)
+      .thru(value => {
+        if (_.isEmpty(value)) {
+          return [];
+        }
+        return value.split(',');
+      })
+      .value(),
+  };
+}
+
+export function stringifyPolicy(policy: PolicyObject): string {
+  return `${policy.timePolicy}${_.isEmpty(policy.aggFunctions)
+    ? ''
+    : `|${policy.aggFunctions.join(',')}`}`;
+}
+
+export function PolicyEditor(props: Props) {
+  const {
+    value = _.first(POLICIES),
+    onDeleteClicked = noop => noop,
+    showDelete = false,
+    onChange = noop => noop,
+    typeTag = '',
+    policyList = POLICIES,
+  } = props;
+  const policy = parsePolicy(value);
+  const handleChange = (field, newValue) => {
+    onChange(
+      stringifyPolicy({
+        ...policy,
+        [field]: newValue,
+      }),
+    );
+  };
+  const showAggs = typeTag === 'timer';
+  let policies = policyList;
+  if (!_.includes(['gauge', 'counter'], typeTag)) {
+    policies = policies.filter(p => !p.startsWith('10m'));
+  }
+
+  return (
+    <div className="clearfix p1 relative" style={{}}>
+      {showDelete && (
+        <a
+          className="absolute"
+          style={{right: 10, top: 10, zIndex: 100}}
+          onClick={onDeleteClicked}>
+          <Icon type="close-circle-o" />
+        </a>
+      )}
+      <div className="col col-6 px1">
+        <label className="block pb1">Resolution:Retention Period</label>
+        <Select
+          placeholder="Select a policy"
+          value={policy.timePolicy}
+          autoComplete="off"
+          onChange={e => {
+            handleChange('timePolicy', e);
+          }}
+          style={{width: '100%'}}
+          notFoundContent={false}>
+          {policies.map(opt => <Option key={opt}>{opt}</Option>)}
+        </Select>
+      </div>
+      <div className="col col-6" style={{opacity: showAggs ? 1 : 0.5}}>
+        <label className="block pb1">Aggregation Functions</label>
+        {showAggs ? (
+          <Select
+            placeholder="Use default if none selected"
+            value={policy.aggFunctions}
+            onChange={e => {
+              handleChange('aggFunctions', e);
+            }}
+            autoComplete="off"
+            mode="tags"
+            style={{width: '100%'}}
+            tokenSeparators={[',']}
+            notFoundContent={false}>
+            {AGGREGATION_FUNCTIONS.map(opt => <Option key={opt}>{opt}</Option>)}
+          </Select>
+        ) : (
+          <small style={{color: 'gray'}}>Only supported for timers</small>
+        )}
+      </div>
+    </div>
+  );
+}
+
+function PoliciesEditor(props) {
+  const {value: policies, onChange, typeTag} = props;
+  return (
+    <div
+      className="clearfix p1"
+      style={{
+        border: '1px solid whitesmoke',
+        lineHeight: 'normal',
+      }}>
+      {_.map(policies, (p, i) => {
+        return (
+          <div
+            key={i}
```

Comment:
Ok so it looks like React won't complain if you pass an object, but the [docs](https://reactjs.org/docs/lists-and-keys.html) mention to use a string. I can do a stringify on the object, since the equality check may fail even if the objects are technically the same from a new reference.